### ISSUE TITLE
The calculated gradient is empty.

### ISSUE THREAD

**Issue by wushupei1997** [jacobgil/pytorch-grad-cam#75]
I'm using BaseCAM. When it ran to "grads = self.activations_and_grads.gradients[-1].cpu().data.numpy()[0, :]", I found that self.activations_and_grads.gradients got an empty result, so he gave an error: "list index out of range".So what's the reason for this?

**Comment by jacobgil** [comment]
Hi, I need more details to help answer.
What is the model, and the target_layer used?

**Comment by wushupei1997** [comment]
hello, I use Resnet50, which is "target_layer=model.layer4[-1]. In addition, because I use multiple GPUs during training, I add the following code to my code：
from collections import OrderedDict
new_state_dict = OrderedDict()
for k, v in state_dict.items():
    name = k[7:]  # remove `module.`
    new_state_dict[name] = v
model.load_state_dict(new_state_dict)
target_layer = model.module.layer4[-1]
I'm using the rest according to your example. I wonder if you can help me with this information, I would appreciate it.

**Comment by jacobgil** [comment]
I suspect there is a problem with the 
model.module.layer4[-1] which seems a bit strange.

I will try further tomorrow.

**Comment by jacobgil** [comment]
I couldnt reproduce it.
What model is passed to CAM, is it the same model?
When you do
print(target_layer), what does it print?

If you can write more details, or maybe a code snippet, that would help debug.

**Comment by jacobgil** [comment]
Closing this unless more details can be shared to help debug

**Comment by ChongjianGE** [comment]
@wushupei1997 
Hi, I met the same problems. Have solved the problem?
